code/modules/power/power_exporter.dm

```diff
@@ -0,0 +1,145 @@
+/obj/machinery/power/exporter
+	name = "power exporter"
+	desc = "It exports power for points, points get rewards."
+	icon = 'icons/obj/machines/dominator.dmi'
+	icon_state = "dominator"
+	density = 1
+	anchored = FALSE
+	verb_say = "states"
+	var/drain_rate = 0	// amount of power to drain per tick
+	var/power_drained = 0 		// has drained this much power
+	var/rating = ""
+	var/active = FALSE
+	var/rewarded = FALSE
+
+/obj/machinery/power/exporter/examine(mob/user)
+	..()
+	if(!active)
+		user << "<span class='notice'>The exporter seems to be offline.</span>"
+	else
+		user << "<span class='notice'>The [src] is exporting [drain_rate] kilowatts of power, it has consumed [power_drained] kilowatts so far.</span>"
+
+obj/machinery/power/exporter/Initialize()
+	..()
+	power_exporter_list += src
+
+obj/machinery/power/exporter/Destroy()
+	power_exporter_list -= src
+	return ..()
+
+/obj/machinery/power/exporter/attackby(obj/item/O, mob/user, params)
+	if(!active)
+		if(istype(O, /obj/item/weapon/wrench))
+			if(!anchored && !isinspace())
+				connect_to_network()
+				user << "<span class='notice'>You secure the [src] to the floor.</span>"
+				anchored = TRUE
+			else if(anchored)
+				disconnect_from_network()
+				user << "<span class='notice'>You unsecure and disconnect the [src].</span>"
+				anchored = FALSE
+			playsound(loc, 'sound/items/Deconstruct.ogg', 50, 1)
+			return
+	return ..()
+
+/obj/machinery/power/exporter/attack_hand(mob/user)
+	..()
+	if (!anchored)
+		user << "<span class='warning'>This device must be anchored by a wrench!</span>"
+		return
+	if (power_drained >= 10000000 && !rewarded) // 10 billion watts
+		rewarded = TRUE
+		visible_message("<span class='danger'>You have been blessed by the gods of engineering for your massive power exports!</span>")
+		new /obj/effect/overlay/temp/explosion/fast(get_turf(src))
+		playsound(src, 'sound/effects/pray_chaplain.ogg', 100, 1)
+		for(var/i in 1 to 3)
+			new /obj/item/clothing/head/orange_beret(get_turf(src))
+			new /obj/item/clothing/suit/poncho/engiponcho(get_turf(src))
+			new /obj/item/clothing/gloves/krav_maga/engi(get_turf(src))
+	interact(user)
+
+/obj/machinery/power/exporter/attack_ai(mob/user)
+	interact(user)
+
+/obj/machinery/power/exporter/attack_paw(mob/user)
+	interact(user)
+
+/obj/machinery/power/exporter/interact(mob/user)
+	if (!anchored)
+		user << "<span class='warning'>This device must be anchored by a wrench!</span>"
+		return
+	if(!Adjacent(user))
+		if(!isAI(user))

```

Comment:
Why wouldn't this just be `if(!Adjacent(user) && !isAI(user))`? I'm not too up-to-date with Byond code, so sorry if that would obviously be wrong or something.